gui/src/app/FileEditor/TextEditor.tsx

```diff
@@ -186,39 +173,95 @@ const toMonacoMarkerSeverity = (
   }
 };
 
+type ToolbarProps = {
```

Comment:
If practical, I think it'd be neater for the `Toolbar` stuff to be in a separate file--it's less distracting compared to combining in the `TextEditor` component file imo.